lib/Hakyll/Web/CompressCss.hs

```diff
@@ -43,6 +45,21 @@ compressSeparators str
     stripSecond = or $ map (isOfPrefix str) $ map (\c -> c ++ " ") separators
     separators  = [" ", "{", "}", ":", ";", ",", ">", "+", "!"]
 
+-- | Compresses separators when starting inside calc().
+compressCalcSeparators :: Int -> String -> String
+compressCalcSeparators 0 str = compressSeparators str
+compressCalcSeparators depth str
+  | stripFirst = compressCalcSeparators depth (tail str)
+  | stripSecond = compressCalcSeparators depth (head str : (drop 2 str))
+  | '(' == head str = '(' : compressCalcSeparators (depth + 1) (tail str)
```

Comment:
Could use something like:

```
   | ('(' : xs) <- str = '(' : compressCalcSeparators (depth + 1) xs 
```

to avoid the use of partial functions?  This also applies to the `')' == head str` line.

Reply:
Ok, uploading a corrected version. Had to add `{-# LANGUAGE PatternGuards #-}` to suppress warnings.